management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -0,0 +1,215 @@
+import org.ehcache.management.providers.statistics.EhcacheStatisticsProvider;
+import org.ehcache.spi.ServiceProvider;
+import org.ehcache.spi.service.CacheManagerProviderService;
+import org.ehcache.spi.service.ExecutionService;
+import org.ehcache.spi.service.ServiceDependencies;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terracotta.management.context.Context;
+import org.terracotta.management.context.ContextContainer;
+import org.terracotta.management.message.DefaultMessage;
+import org.terracotta.management.notification.ContextualNotification;
+import org.terracotta.management.registry.MessageConsumer;
+import org.terracotta.management.registry.StatisticQuery;
+import org.terracotta.management.stats.ContextualStatistics;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.ehcache.internal.executor.ExecutorUtil.shutdownNow;
+
+
+/**
+ * @author Mathieu Carbou
+ */
+@ServiceDependencies({CacheManagerProviderService.class, ManagementRegistryService.class, ExecutionService.class, TimeSourceService.class})
+public class DefaultCollectorService implements CollectorService, CacheManagerListener {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCollectorService.class);
+
+  private final AtomicBoolean collectingStats = new AtomicBoolean();
+  private volatile ScheduledFuture<?> task;
+
+  private final ConcurrentMap<String, StatisticQuery.Builder> selectedStatsPerCapability = new ConcurrentHashMap<String, StatisticQuery.Builder>();
+  private final MessageConsumer messageConsumer;
+
+  private volatile TimeSourceService timeSource;
+  private volatile ManagementRegistryService managementRegistry;
+  private volatile ScheduledExecutorService scheduledExecutorService;
+  private volatile CacheManagerProviderService cacheManagerProviderService;
+  private volatile ManagementRegistryServiceConfiguration configuration;
+
+  public DefaultCollectorService(MessageConsumer messageConsumer) {
+    this.messageConsumer = messageConsumer;
+  }
+
+  @Override
+  public void start(ServiceProvider serviceProvider) {
+    timeSource = serviceProvider.getService(TimeSourceService.class);
+    managementRegistry = serviceProvider.getService(ManagementRegistryService.class);
+    configuration = managementRegistry.getConfiguration();
+    cacheManagerProviderService = serviceProvider.getService(CacheManagerProviderService.class);
+    scheduledExecutorService = serviceProvider.getService(ExecutionService.class).getScheduledExecutor(DefaultCollectorService.class.getSimpleName());
```

Comment:
ok! I've changed what was there before because of these 2 reasons:

1. because I ended to have some threads with null in names since the alias is null by default.

    `return new Thread(r, "Ehcache [" + alias + "]-" + threadCount.getAndIncrement());`

2. also because I wanted to separate the thread pool responsible of computing statistics (in the  `EhcacheStatisticsProvider` class and the thread pool responsible for collecting and pushing messages (`DefaultCollectorService`)

=> What would be the best way to fix that @chrisdennis ? I need to redo and thus have 2 configurable aliases in the `ManagementRegistryServiceConfiguration` ? Thanks :-)